src/RenderMethods/RenderFieldTemplate.tsx

```diff
@@ -1,150 +1,80 @@
-          (key: string) => key === currentField,
-        );
-        newUiSchema[currentSection]['ui:order'].splice(currentSectionIndex + 1, 0, newKey);
-      } else {
-        newUiSchema[currentSection] = {};
-        newUiSchema[currentSection]['ui:order'] = [newKey];
-      }
-    }
-    setUiSchema(newUiSchema);
-  }
-
-  function addItem() {
-    const newKey = makeid(10);
-    addItemToSchema(newKey);
-    addItemToUiSchemaOrder(newKey);
-  }
-
-  function addSection() {
-    const newSchema = { ...vernaSchema };
+  function addItem(widgetProps?: ShowCaseWidgetProps) {
     const newKey = makeid(10);
-    if (newSchema.properties) {
-      newSchema.properties[newKey] = sectionDefinition(newKey);
-    }
-    setSchema(newSchema);
-    const currentSection = getCurrentSection(id);
-    const newUiSchema = { ...uiSchema };
-    if (newUiSchema['ui:order']) {
-      const currentSectionIndex =
-        newUiSchema['ui:order'].findIndex((key) => key === currentSection) || 0;
-      newUiSchema['ui:order'].splice(currentSectionIndex + 1, 0, newKey);
-      setUiSchema(newUiSchema);
-    }
+    addItemToSchema(newKey, id, verna, widgetProps?.type);
+    addItemToUiSchema(newKey, id, verna, widgetProps?.widgetName);
   }
 
-  function removeItem() {
-    const newSchema = { ...vernaSchema };
-    const newUiSchema = { ...uiSchema };
-    if (selector) {
-      if (newSchema.properties) delete newSchema.properties[currentField];
-      if (newUiSchema['ui:order']) {
-        newUiSchema['ui:order'] = newUiSchema['ui:order'].filter(
-          (key: string) => key !== currentField,
-        );
-      }
-    } else {
-      if (newSchema.properties) {
-        const sectionProperties = newSchema.properties[currentSection] as JSONSchema7;
-        if (sectionProperties.properties) {
-          delete sectionProperties.properties[currentField];
-        }
-      }
-      if (newUiSchema[currentSection]['ui:order']) {
-        newUiSchema[currentSection]['ui:order'] = newUiSchema[currentSection]['ui:order'].filter(
-          (key: string) => key !== currentField,
-        );
-      }
-    }
-    setSchema(newSchema);
-    setUiSchema(newUiSchema);
+  function onDrop(event: React.DragEvent<HTMLDivElement>) {
+    event.preventDefault();
+    const widgetProps = JSON.parse(event.dataTransfer.getData('object'));
+    addItem(widgetProps);
+    setDraggingOver(false);
   }
 
-  function removeSection() {
-    const newSchema = { ...vernaSchema };
-    const newUiSchema = { ...uiSchema };
-    if (newSchema.properties) {
-      delete newSchema.properties[currentSection];
-    }
-    setSchema(newSchema);
-    delete newUiSchema[currentSection];
-    setUiSchema(newUiSchema);
+  function onDragOver(event: React.DragEvent<HTMLDivElement>) {
+    event.preventDefault();
+    setDraggingOver(true);
   }
 
   return (
     <div>
-      {id}
       {children}
+      {verna.isEditor && (
```

Comment:
I wonder if instead to check the value of `isEditor` here it should not be better to use this custom component`RenderFieldTemplate` only when `isEditor` is true. What do you think?